pymongo/database.py

```diff
@@ -541,36 +541,39 @@ def command(self, command, value=1, check=True,
                                  check, allowable_errors, read_preference,
                                  codec_options, session=session, **kwargs)
 
-    def _list_collections(self, sock_info, slave_okay, criteria=None,
-                          session=None):
+    def list_collections(self, sock_info, slave_okay, filter=None,
+                         session=None, batch_size=0):
         """Internal listCollections helper."""
-        criteria = criteria or {}
-        cmd = SON([("listCollections", 1), ("cursor", {})])
-        if criteria:
-            cmd["filter"] = criteria
+        filter = filter or {}
+        cmd = SON([("listCollections", 1),
+                   ("cursor", {"batchSize": batch_size}),
+                   ("filter",filter)])
 
         if sock_info.max_wire_version > 2:
             coll = self["$cmd"]
             with self.__client._tmp_session(session, close=False) as s:
                 cursor = self._command(
                     sock_info, cmd, slave_okay, session=s)["cursor"]
                 return CommandCursor(coll, cursor, sock_info.address, session=s,
-                                     explicit_session=session is not None)
+                                     explicit_session=session is not None, batch_size=batch_size)
         else:
             coll = self["system.namespaces"]
+            if "name" in filter:
+                filter["name"] = coll.database.name + "." + filter["name"]
```

Comment:
Nice, but you also need to validate that `filter["name"]` is a string. A good error might be:
```python
if not isinstance(filter["name"], string_type):
    raise TypeError("filter['name'] must be a string on MongoDB 2.6")
```
https://github.com/mongodb/specifications/blob/master/source/enumerate-collections.rst#filters